src/batchUploader.ts

```diff
@@ -129,21 +131,68 @@ export class BatchUploader {
         return offlineStoragePercentage >= rampNumber;
     }
 
+    private shouldDebounceAST(): boolean {
+        const now = Date.now();
+        if (now - this.lastASTEventTime < this.AST_DEBOUNCE_MS) {
+            return true;
+        }
+
+        this.lastASTEventTime = now;
```

Comment:
At first glance like it's just doing a check but it's actually doing a side effect. I personally would modify `lastASTEventTime` outside of the function. If you need this side effect in the function, I would rename the function to `checkLastDebounceASTAndUpdate()` or something similar.

Reply:
looks like i actually do need this here ebcause if i try to call `shouldDebounceAST` in a unit test without it, it won't have the lastASTEventTime and it will always return `false`.  so i'll add this back and rename